fipy/viewers/matplotlibViewer/matplotlib2DGridViewer.py

```diff
@@ -90,19 +82,24 @@ def _getSuitableVars(self, vars):
         # this viewer can only display one variable
         return [vars[0]]
 
+    def _make_mappable(self):
+        xmin, ymin = self.vars[0].mesh.extents['min']
+        xmax, ymax = self.vars[0].mesh.extents['max']
+
+        self.image = self.axes.imshow(self._data,
+                                      extent=(xmin, xmax, ymin, ymax),
+                                      norm=self.norm,
+                                      cmap=self.cmap)
+        return self.image
+
     @property
     def _data(self):
         from fipy.tools.numerix import array, reshape
         return reshape(array(self.vars[0]), self.vars[0].mesh.shape[::-1])[::-1]
 
     def _plot(self):
-        self.norm.vmin = self._getLimit(('datamin', 'zmin'))
-        self.norm.vmax = self._getLimit(('datamax', 'zmax'))
-
-        self.image.set_data(self.norm(self._data))
-
-        if self.colorbar is not None:
-            self.colorbar.plot()
+        super(Matplotlib2DGridViewer, self)._plot()
```

Comment:
Haven't seen `super` before... [apparently an inheritance trick](https://stackoverflow.com/questions/222877/what-does-super-do-in-python) to help other classes use this one's `_plot` method?

Reply:
I don't think that's what super does. I may misunderstand what you intended with your statement though. Super allows the sub-class (where it's used) to not require reference to the parent class by name. If the inheritance tree changes or the method changes position in the inheritance (it might move up the tree for example) tree then there's no need to change the code that calls the inherited method. That's the main benefit.

Reply:
That's actually subtly wrong. I should amend that to say if the parent class name changes then there's no need to change all the calls to the parent class method. That is all it does.